collections/nemo_asr/nemo_asr/audio_preprocessing.py

```
@@ -0,0 +1,470 @@
+        window_size: Size of window for fft in seconds. Used to calculate the
+            win_length arg for mel spectrogram, overrides mel_kwargs if set.
+            Defaults to None
+        window_stride: Stride of window for fft in seconds. Used to caculate
+            the hop_length arg for mel spect, overrides mel_kwargs if set.
+            Defaults to None
+        window: Windowing function for fft. can be one of ['hann',
+            'hamming', 'blackman', 'bartlett', None]. Overrides mel_kwargs
+            if set; otherwise, torchaudio defaults to hann.
+            Defaults to None
+        n_mfcc: Number of coefficients to retain
+            Defaults to 64
+        dct_type: Type of discrete cosine transform to use
+        norm: Type of norm to use
+        log: Whether to use log-mel spectrograms instead of db-scaled.
+            Defaults to True.
+        mel_kwargs: Dict of arguments for torchaudio.transforms.MelSpectrogram
+    """
+    def __init__(
+            self, *,
+            sample_rate=16000,
+            window_size=None,
+            window_stride=None,
+            window=None,
+            n_mfcc=64,
+            dct_type=2,
+            norm='ortho',
+            log=True,
+            mel_kwargs=None,
+            **kwargs):
+        super().__init__(**kwargs)
+
+        if mel_kwargs is None:
+            mel_kwargs = {}
+
+        # Use the sample rate given instead of in mel_kwargs
+        if 'sample_rate' in mel_kwargs:
+            del mel_kwargs['sample_rate']
+
+        # Override mel_kwargs if window_size or window_stride are set
+        if window_size is not None:
+            mel_kwargs['win_length'] = int(sample_rate * window_size)
+        if window_stride is not None:
+            mel_kwargs['hop_length'] = int(sample_rate * window_stride)
+
+        # Set window_fn if window arg is given
+        if window:
+            window_fn = self.torch_windows.get(window, None)
+            if window_fn is None:
```

Same comment as above.
A None window type should be allowed. AKA your window is just 1.
However, an unknown window type should throw this error. This is a stuble difference depending on what you intend self.torch_windows["none"] to mean.

---

Done (see above)